cli/src/commands/extensions/extension.rs

```diff
@@ -0,0 +1,143 @@
+use std::{convert::TryFrom, fs::File, io::Read, path::PathBuf};
+
+use anyhow::{anyhow, Result};
+use serde::Deserialize;
+
+const MANIFEST_NAME: &str = "PhylumExt.toml";
+
+#[derive(Debug)]
+pub struct Extension {
+    path: PathBuf,
+    manifest: ExtensionManifest,
+}
+
+#[derive(Deserialize, Debug)]
+pub struct ExtensionManifest {
+    name: String,
+    description: Option<String>,
+    entry_point: String,
+}
+
+impl Extension {
+    pub fn name(&self) -> &str {
+        &self.manifest.name
+    }
+
+    pub fn description(&self) -> Option<&str> {
+        self.manifest.description.as_deref()
+    }
+
+    /// Install the extension in the default path.
+    pub fn install(&self) -> Result<()> {
+        println!("Installing extension {}...", self.name());
+        let target_prefix = extensions_path()?.join(self.name());
+
+        if target_prefix == self.path {
+            return Err(anyhow!(
+                "extension path and installation path are identical, skipping"
+            ));
+        }
+
+        for entry in walkdir::WalkDir::new(&self.path) {
```

Comment:
```suggestion
        for entry in WalkDir::new(&self.path) {
```

Should import WalkDir here.